getgauge/static_loader.py

```diff
@@ -18,9 +19,7 @@ def reload_steps(file_path, content=None):
 
 def load_files(step_impl_dirs):
     for step_impl_dir in step_impl_dirs:
-        for dirpath, _, files in os.walk(step_impl_dir):
-                py_files = (os.path.join(dirpath, f) for f in files if f.endswith('.py'))
-                for file_path in py_files:
-                        pf = Parser.parse(file_path)
-                        if pf:
-                                load_steps(pf)
+        for file_path in glob.glob(f"{step_impl_dir}/**/*.py", recursive=True):
+            pf = Parser.parse(file_path)
```

Comment:
Are globbing and walking semantically equivalent? Worth noting that despite its ugliness, there is some history here around supporting relative imports you might want to look through the history for, including #360 and #364 (although maybe not directly related)

Reply:
Both ways are doing the same thing, if you use "glob" with "recursive=True". In general you have more control which files/dirs to consider using `os.walk()` although we are interested in one single file type only. Since we are interested in Python files only, I think the better way is `glob()`!

Also the steps outside of the usual workspace is fine, as the root directory is used still.

Reply:
Sometimes such approaches differ in how they handle symlinks and such, and although I'm not sure that is relevant here - is worth asking the question (I'm not a Python expert)